src/components/DownloadTrajectoryMenu/index.tsx

```diff
@@ -0,0 +1,87 @@
+import React from "react";
+import { Button, Tooltip } from "antd";
+import { ISimulariumFile } from "@aics/simularium-viewer/type-declarations";
+
+import { DATA_BUCKET_URL, TOOLTIP_COLOR } from "../../constants";
+import {
+    NetworkedSimFile,
+    LocalSimFile,
+    isNetworkSimFileInterface,
+} from "../../state/trajectory/types";
+import { Download } from "../Icons";
+
+import styles from "./style.css";
+
+interface DownloadTrajectoryMenuProps {
+    isBuffering: boolean;
+    simulariumFile: LocalSimFile | NetworkedSimFile;
+}
+
+const DownloadTrajectoryMenu = ({
+    isBuffering,
+    simulariumFile,
+}: DownloadTrajectoryMenuProps): JSX.Element => {
+    const [isDownloading, setIsDownloading] = React.useState(false);
+    const fileIsLoaded = () => !!simulariumFile.name;
+
+    const getHref = () => {
+        if (!fileIsLoaded()) {
+            return "";
+        }
+        if (isNetworkSimFileInterface(simulariumFile)) {
+            return `${DATA_BUCKET_URL}/trajectory/${simulariumFile.name}`;
+        } else {
+            const data: ISimulariumFile = simulariumFile.data;
+            // won't work if they loaded a binary file
+            const blob = data.getAsBlob();
+            return URL.createObjectURL(blob);
+        }
+    };
+
+    const downloadFile = async (fileName: string): Promise<void> => {
+        const downloadLink = document.createElement("a");
+        setIsDownloading(true);
```

Comment:
have you seen the button switch to "Downloading..."? Because none of these are written as async functions, it seems like the time between `setIsDownloading(true)` and `setIsDownloading(false)` is always going to be imperceivable